app/views/warehouses/_form.html.erb

```diff
@@ -0,0 +1,32 @@
+<%= form_with(model: warehouse, local: true) do |form| %>
+  <% if warehouse.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(warehouse.errors.count, "error") %> prohibited this warehouse from being saved:</h2>
```

Comment:
it'd be cool to have at least a partial to render errors, so you can re-use this logic in other catalogs

<%= render partial: 'shared/errors', locals: { errors: warehouse.errors } %>